cmd/api-linter/cli.go

```diff
@@ -183,25 +184,31 @@ func (c *cli) lint(rules lint.RuleRegistry, configs lint.Configs) error {
 		Reporter:       rep,
 	}
 
-	// Compile files.
-	files, err := compiler.Compile(context.Background(), c.ProtoFiles...)
-
-	// After compilation, check if the handler collected any errors.
-	// This is the primary source of truth for parse errors when using a
-	// custom reporter that continues on error.
-	if len(collectedErrors) > 0 {
-		errorStrings := make([]string, len(collectedErrors))
-		for i, e := range collectedErrors {
-			errorStrings[i] = e.Error()
+	var compiledFiles linker.Files
+	for _, protoFile := range c.ProtoFiles {
+		// Compile each file individually.
```

Comment:
Nit: Why? The code is self-explanatory as far as what it is doing, the comment should be providing rationale.

https://testing.googleblog.com/2017/07/code-health-to-comment-or-not-to-comment.html